subprojects/android-test/test-screenshot/src/main/java/com/avito/android/test/screenshot_test/internal/ViewSaver.kt

```diff
@@ -0,0 +1,59 @@
+package com.avito.android.test.screenshot_test.internal
+
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.os.Build
+import android.os.Handler
+import android.view.PixelCopy
+import android.view.View
+import com.avito.android.test.screenshot_test.test.IdlieableActivity
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+internal fun View.saveScreenshot(activity: IdlieableActivity, name: String) {
+    val bitmap = Bitmap.createBitmap(width,
+        height,
+        Bitmap.Config.ARGB_8888)
+    val locationOfViewInWindow = IntArray(2)
+    val window = activity.window
+    activity.countingIdlingResource.increment()
+    getLocationInWindow(locationOfViewInWindow)
+    try {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            PixelCopy.request(
+                window,
+                Rect(
+                    locationOfViewInWindow[0],
+                    locationOfViewInWindow[1],
+                    locationOfViewInWindow[0] + width,
+                    locationOfViewInWindow[1] + height
+                ), bitmap, { copyResult ->
+                if (copyResult == PixelCopy.SUCCESS) {
+                    saveToSdCard(name, bitmap)
+                }
+                activity.countingIdlingResource.decrement()
+            },
+                Handler(activity.mainLooper)
+            )
+        }
+    } catch (e: IllegalArgumentException) {
+        e.printStackTrace()
+        activity.countingIdlingResource.decrement()
+    }
+}
+
+private fun View.saveToSdCard(name: String, bitmap: Bitmap) {
```

Comment:
I don't get why it's an extension function on `View` type

Reply:
because I'm too lazy to pass it as regular argument and use "view.width, view.height, etc" all the time

Reply:
I don't see usages of `width` and `height`